tsp_simple.cc

```diff
@@ -235,12 +243,26 @@ bool RouteBuilder(const TSPTWDataDT& data, RoutingModel& routing,
   return routing.RoutesToAssignment(routes, true, false, assignment);
 }
 
-std::vector<std::vector<IntervalVar*>> RestBuilder(const TSPTWDataDT& data,
-                                                   RoutingModel& routing) {
+std::vector<std::vector<IntervalVar*>>
+RestBuilder(const TSPTWDataDT& data, RoutingModel& routing, const int64 horizon) {
   Solver* solver          = routing.solver();
   const int size_vehicles = data.Vehicles().size();
   std::vector<std::vector<IntervalVar*>> stored_rests;
+
+  // TODO: only the last pause is pushed towards the end of route. If there are multiple
+  // pauses per vehicle then multiple "rest" dimensions are needed.
+  routing.AddDimension(/*zero_evaluator*/ 0, horizon, horizon, true, kRestPosition);
+
   for (int vehicle_index = 0; vehicle_index < size_vehicles; ++vehicle_index) {
+    RoutingDimension* rest_dimension = routing.GetMutableDimension(kRestPosition);
+    // Add a small cost so that pause will be pushed towards the end of the route if it
+    // doesn't increase other costs. Note: the cost coefficient is actually very small
+    // (1 / CUSTOM_BIGNUM) since every other cost is multipled with CUSTOM_BIGNUM
+    rest_dimension->SetSpanCostCoefficientForVehicle(1, vehicle_index);
+    rest_dimension->CumulVar(routing.Start(vehicle_index))->SetMax(0);
+
+    IntVar* const end_rest_cumul_var = rest_dimension->CumulVar(vehicle_index);
```

Comment:
Isn't it the cumul var of a node or a starting node ? Which variable are you trying to manipulate ?

Reply:
It should have been `rest_dimension->CumulVar(routing.End(vehicle_index));` weird that it returns expected results like this. gonna force-push with fix

Reply:
Apparently it wasn't surprising that it was working with just `vehicle_index` since pushing the very first node has the same effect of pushing the last node and it would work as long as there are more services than vehicles -- so it would be a hell to debug it!